pkg/router/crl/crl.go

```diff
@@ -11,80 +11,275 @@ import (
 	"io/ioutil"
 	"net/http"
 	"os"
+	"path/filepath"
 	"strings"
 	"time"
 
 	logf "github.com/openshift/router/log"
+	"github.com/openshift/router/pkg/util"
 	kerrors "k8s.io/apimachinery/pkg/util/errors"
 )
 
 var log = logf.Logger.WithName("crl")
 
-const crlFilePermissions = 0600
+const (
+	// crlFilePermissions is the permission bits used by the CRL file.
+	crlFilePermissions = 0660
+	// crlFallbackTime is how long to wait before retrying if nextUpdate can't be determined
+	crlFallbackTime = 5 * time.Minute
+	// errorBackoffTime is how long to wait before retrying if a generic error happens during CRL refresh
+	errorBackoffTime = 5 * time.Minute
+	// mtlsBaseDirectory is the directory where all the files and temp directories created here will live
+	mtlsBaseDirectory = "/var/lib/haproxy/mtls/"
+	// crlBasename is the name of the crl file created here
+	crlBasename = "crls.pem"
+	// caBundleBasename is the name of the CA bundle "created" here (really just copied from the CA bundle configmap)
+	caBundleBasename = "ca-bundle.pem"
+	// directoryReaperTimer is how often the stale directory reaper runs
+	directoryReaperTimer = 5 * time.Minute
+)
+
+var (
+	// mtlsLatestSymlink is the fully qualified path to the symlink used in the template
+	mtlsLatestSymlink = filepath.Join(mtlsBaseDirectory, "mtls-latest")
+	// mtlsLatestSymlink is the fully qualified path to the staging symlink, used to atomically replace the existing mtlsLatestSymlink
+	mtlsLatestStagingSymlink = filepath.Join(mtlsBaseDirectory, "mtls-latest.new")
+	// CRLFilename is the fully qualified path to the currently in use crl file
+	CRLFilename = filepath.Join(mtlsLatestSymlink, crlBasename)
+	// CRLFilename is the fully qualified path to the currently in use CA bundle
+	CABundleFilename = filepath.Join(mtlsLatestSymlink, caBundleBasename)
+)
 
 // authorityKeyIdentifier is a certificate's authority key identifier.
 type authorityKeyIdentifier struct {
 	KeyIdentifier []byte `asn1:"optional,tag:0"`
 }
 
+// authorityKeyIdentifierOID is the ASN.1 object identifier for the authority key identifier extension.
 var authorityKeyIdentifierOID = asn1.ObjectIdentifier{2, 5, 29, 35}
 
-func ManageCRLs(caBundleFilename, crlFilename string, caUpdateChannel <-chan struct{}) error {
-	nextUpdate, err := updateCRLFile(caBundleFilename, crlFilename)
-	if err != nil {
-		return err
+// ManageCRLs reads the file at caBundleFilename, ensures that the file at crlFilename contains all specified crls, and
+// updates any crls that have expired. ManageCRLs does this synchronously initially, then spins off a goroutine to
+// update CRLs when necessary in the future. Returns a boolean that's true if any of the certificates in the CA bundle
```

Comment:
Might as well mention both goroutines.
```suggestion
// updates any CRLs that have expired. ManageCRLs does this synchronously initially, then spins off a goroutine to
// update CRLs when necessary in the future and another goroutine to reap stale CRLs. Returns a Boolean that's
// true if any of the certificates in the CA bundle
```